Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -546,6 +557,15 @@ void FullscreenManager::dispatchFullscreenChangeEvents()
     m_fullscreenChangeEventTargetQueue.swap(changeQueue);
     Deque<GCReachableRef<Node>> errorQueue;
     m_fullscreenErrorEventTargetQueue.swap(errorQueue);
+
+    if (m_pendingPromise) {
+        if (!errorQueue.isEmpty())
+            m_pendingPromise->reject(Exception { TypeError });
+        else if (!changeQueue.isEmpty())
+            m_pendingPromise->resolve();
+        m_pendingPromise = nullptr;
```

Comment:
What about the case where both errorQueue and changeQueue are empty? Isn't it possible? If it isn't then it should be an assertion.

If it is possible, then I think we have an issue because you'll clear the pendingPromise pointer without resolving it, which would not be OK.